core/utilities.lua

```diff
@@ -427,6 +427,63 @@ utilities.utf8_to_utf16le = function (str)
   return ustr
 end
 
+local icu = require("justenoughicu")
+
+local icuFormats = function (format)
+  if format == "roman" then return "romanlow" end
+  if format == "Roman" then return "roman" end
+  return format
+end
+
+local icuFormat = function (num, format)
+  local ok, result  = pcall(function() return icu.format_number(num, format) end)
+  return tostring(ok and result or num)
+end
+
+-- Language specific number formatters add functions to this table,
+-- see e.g. languages/tr.lua
+utilities.formatNumber = {
+  und = {
+
+    alpha = function (num)
+      local out = ""
+      local a = string.byte("a")
+      repeat
+        num = num - 1
+        out = string.char(num % 26 + a) .. out
+        num = (num - num % 26) / 26
+      until num < 1
+      return out
+    end
+
+  }
+}
+
+setmetatable (utilities.formatNumber, {
+    __call = function (self, num, format, case)
+      if not case then
+        if format:match("^%l") then
+          case = "lower"
+        elseif format:match("^.%l") then
+          case = "title"
+        else
+          case = "upper"
+        end
+      end
+      local lang = SILE.settings.get("document.language")
+      local format = format:lower()
+      local result
+      if self[lang] and type(self[lang][format]) == "function" then
+        result = self[lang][format](num)
+      elseif type(self["und"][format]) == "fuction" then
```

Comment:
`function`. Otherwise, I love it.

Reply:
Oopsie. Corrected, good catch.

Reply:
Also, if your tests passed with that misspelling then either the tests are wrong or there’s a case that was not tested...

Reply:
The test does not currently have an `.expected` file at all. I didn't add one to this PR either because I don't have the same font versions as Travis and the local testing PR isn't quite fully baked yet.

Reply:
Added `alpha` to test, and used `ROMAN` so that we test all three case modes too. Still no expect for reasons noted.